seacrowd/sea_datasets/ac_iquad/ac_iquad.py

```diff
@@ -0,0 +1,236 @@
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema=f"seacrowd_{SEACROWD_SCHEMA_NAME}",
+            subset_id=f"{_DATASETNAME}_complex",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features_dict = {
+                "question": datasets.Value("string"),
+                "sparql": datasets.Value("string"),
+                "answer": datasets.Value("string"),
+                "context": datasets.Value("string"),
+                "answerline": datasets.Value("string"),
+            }
+
+            if self.config.subset_id.split("_")[2] == "complex":
+                features_dict["type"] = datasets.Value("string")
+
+            features = datasets.Features(features_dict)
+
+        elif self.config.schema == f"seacrowd_{self.SEACROWD_SCHEMA_NAME}":
+            features = schemas.qa_features
+
+            if self.config.subset_id.split("_")[2] == "complex":
+                features["meta"] = {"sparql": datasets.Value("string"), "answer_meta": datasets.Value("string"), "type": datasets.Value("string")}
+
+            else:
+                features["meta"] = {"sparql": datasets.Value("string"), "answer_meta": datasets.Value("string")}
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        subset = self.config.name.split("_")[2]
+        data_dir = dl_manager.download_and_extract(_URLS[_DATASETNAME])
+
+        if subset == "simple":
+            subset = "single"
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, f"{subset}_train.json"),
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, f"{subset}_test.json"),
+                    "split": "test",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+
+        with open(filepath, "r", encoding="utf-8") as file:
+            data_json = json.load(file)
+
+        df = pd.json_normalize(data_json)
+
+        for index, row in df.iterrows():
+
+            if self.config.schema == "source":
+                example = row.to_dict()
+
+                if self.config.subset_id.split("_")[2] == "complex":
+                    example["type"] = row["tipe"]
```

Comment:
Since you're keeping the example value as-is without altering or removing `tipe` value, it will throw a `KeyError` on example generation. Consider doing this in L213:

```suggestion
            if self.config.schema == "source":
                example = row.to_dict()

                if self.config.subset_id.split("_")[2] == "complex":
                    example["type"] = example.pop("tipe")
```
so that the `tipe` key is removed and renamed accordingly to `type`, aligned to the features that you defined on `_info` method.

Reply:
I don't see any `KeyError` while generating an example on that particular (`source`) schema. Because in my `_info` method, I explicitly call my variable as a `type` (not a `tipe`), so it will not cause any error.

This is my `_info` method regarding the `complex` type.
```
if self.config.subset_id.split("_")[2] == "complex":
    features_dict["type"] = datasets.Value("string")
```
This is the screen-capture while generating an example (will not cause any error at all).
![image](https://github.com/SEACrowd/seacrowd-datahub/assets/78993021/3c06671d-1412-4b05-b111-d8019a1cf872)

Reply:
This is on the `seacrowd_qa` schema, instead of `source` schema below.
![image](https://github.com/SEACrowd/seacrowd-datahub/assets/78993021/a06b10f3-5bf0-46ce-a075-c2be4bfef7a3)

Reply:
But, for the sake of the clean-code concept, I propose to change that line to:

```
example["type"] = example.pop("tipe", None)
```

Done in https://github.com/SEACrowd/seacrowd-datahub/pull/641/commits/2bfd4630eca7ad8e5ea845287eeab73f7976fc42 commit. Thanks for the suggestion, Sir!

Reply:
Hmm, are we probably using different versions of `datasets`? Or did you run the SEACrowd test-checking script using this?

```
python -m tests.test_seacrowd seacrowd/sea_datasets/ac_iquad/ac_iquad.py --subset_id ac_iquad_complex
```

Reply:
Yes, I use these scripts to test the dataloader.
```
python -m tests.test_seacrowd seacrowd/sea_datasets/ac_iquad/ac_iquad.py --subset_id ac_iquad_simple
python -m tests.test_seacrowd seacrowd/sea_datasets/ac_iquad/ac_iquad.py --subset_id ac_iquad_complex
```

This is the screen capture for both test results.

This is for `simple`.
![image](https://github.com/SEACrowd/seacrowd-datahub/assets/78993021/91d08dad-8aa7-48ab-9d95-ebf9197b9c49)
![image](https://github.com/SEACrowd/seacrowd-datahub/assets/78993021/3e35de9e-4380-495a-b8a8-2d662436bac7)

And for `complex`.
![image](https://github.com/SEACrowd/seacrowd-datahub/assets/78993021/419d844b-f625-46b7-894d-23b2058eb63c)
![image](https://github.com/SEACrowd/seacrowd-datahub/assets/78993021/78fef9ee-26a0-4b2a-be5b-8ce913f9113f)

Both return an OK response.